### PR TITLE
GPTQ Vision Utils

### PR DESCRIPTION
**Pull request by kylesayrs**
## Purpose ##
* Add utils used by #914

## Changes ##
* Add `DisableQuantization`
* Add `calibration_forward_context`
* Add `CompressionLogger`

## Testing ##
* Added tests in `tests/llmcompressor/utils/test_helpers.py`

### CODE REVIEW

File: src/llmcompressor/utils/metric_logging.py [hunk 1]
```diff
@@ -51,3 +53,50 @@ def get_layer_size_mb(module: Module) -> float:
     total_size_mb = total_size / (1e6)  # Convert bytes to MB
 
     return total_size_mb
+
+
+class CompressionLogger:
```
Comment (dsikka) [inline src/llmcompressor/utils/metric_logging.py:58]: Why is this required given the current logging that already exists in gptq?

Comment (kylesayrs) [reply src/llmcompressor/utils/metric_logging.py:58]: Breaking this out means more real estate is dedicated to describing the algorithm, not logging

File: src/llmcompressor/utils/helpers.py [hunk 2]
```diff
@@ -1080,3 +1084,32 @@ def __enter__(self):
...
+
+    - Remove gradient calculations
+    - Disable the KV cache
+    - Disable quantization from QuantizationModifier
+    """
+    model.eval()
+
+    with (
+        torch.no_grad(),
+        DisableKVCache(model),
```
Comment (dsikka) [inline src/llmcompressor/utils/helpers.py:1112]: do we actually disable the kv cache, anywhere outside of gptq?

Comment (kylesayrs) [reply src/llmcompressor/utils/helpers.py:1112]: No, but technically we should be disabling the kv cache for all calibration forward passes (pipelines), as calibration only consists of prefill and populating the kvcache only costs performance and memory.

Comment (dsikka) [reply src/llmcompressor/utils/helpers.py:1112]: Yeah I was gonna suggest adding this to SparseGPTQ but can do later

**Review by dsikka — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]
